src/sql/src/session/vars.rs

```diff
@@ -1958,6 +1962,29 @@ impl SessionVars {
         }
     }
 
+    /// Sets the "role default" parameter named `name` to the value represented by `value`.
+    ///
+    /// A role default only takes effect at the start of a Session, so running
+    /// `ALTER ROLE ... SET ...` has no visible impact until a new connection is started.
+    pub fn set_role_default(&mut self, name: &str, input: VarInput) -> Result<(), VarError> {
+        let name = UncasedStr::new(name);
+        if name == MZ_VERSION_NAME {
+            Err(VarError::ReadOnlyParameter(MZ_VERSION_NAME.as_str()))
+        } else if name == IS_SUPERUSER_NAME {
+            Err(VarError::ReadOnlyParameter(IS_SUPERUSER_NAME.as_str()))
+        } else if name == MAX_IDENTIFIER_LENGTH.name {
+            Err(VarError::ReadOnlyParameter(
+                MAX_IDENTIFIER_LENGTH.name.as_str(),
+            ))
```

Comment:
Might be best to extract this into it's own function if it's copied across the function above and below. The function below looks like it has slightly different validations, and I can't tell if it's on purpose or accident.

Reply:
Fixed!